### PR TITLE
Stop referencing inspector_service from diagnostics_node to simplify mental model.

### PR DESCRIPTION
**Pull request by polina-c**
This PR is created from https://github.com/flutter/devtools/pull/5224

### CODE REVIEW

**Comment by polina-c** [comment]
@matt-ragonese , can you test custom screen is feeling well with cl/540004219?

**Comment by polina-c** [comment]
G3 testing: cl/540004219

File: packages/devtools_app/lib/src/shared/diagnostics/inspector_service.dart [hunk 1]
```diff
@@ -1037,8 +1042,8 @@ abstract class ObjectGroupBase implements Disposable {
         throw UnimplementedError(
           'getSourcePosition not implemented. $location',
         );
-//        return inspectorLibrary.getSourcePosition(
-//            debugProcess, location.script, location.tokenPos, this);
+        // return inspectorLibrary.getSourcePosition(
+        // debugProcess, location.script, location.tokenPos, this);
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/diagnostics/inspector_service.dart:1046]: do we need these commended out lines or can we remove?

File: packages/devtools_app/lib/src/shared/diagnostics/object_group_api.dart [hunk 2]
```diff
@@ -0,0 +1,43 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/diagnostics/object_group_api.dart:1]: 2023

File: packages/devtools_app/lib/src/shared/diagnostics/object_group_api.dart [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:vm_service/vm_service.dart';
+
+import '../eval_on_dart_library.dart';
+import 'primitives/instance_ref.dart';
+
+abstract class ObjectGroupApi<T extends DiagnosticableTree>
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/diagnostics/object_group_api.dart:11]: add dart doc

Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/diagnostics/object_group_api.dart:11]: Should this class really be `InspectorObjectGroupApi`? It seems all these abstract methods have to do with the inspector and aren't really general

File: packages/devtools_app/lib/src/shared/diagnostics/inspector_service.dart [hunk 4]
```diff
@@ -583,7 +584,8 @@ class InspectorService extends InspectorServiceBase {
   }
 }
 
-abstract class ObjectGroupBase implements Disposable {
+/// This class has additional descenders in Google3.
+abstract class ObjectGroupBase extends ObjectGroupApi<RemoteDiagnosticsNode> {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/diagnostics/inspector_service.dart:588]: same would apply here: `InspectorObjectGroupBase`

**Review by kenzieschmoll — APPROVED** [review]

**Pull request merged**